Sources/SpeziViews/Views/ManagedNavigationStack/ManagedNavigationStack.swift

```diff
@@ -0,0 +1,193 @@
+/// struct Onboarding: View {
+///     @AppStorage("didCompleteOnboarding") var didCompleteOnboarding = false
+///     @State private var localNotificationAuthorization = false
+///
+///     var body: some View {
+///         ManagedNavigationStack(didComplete: $didCompleteOnboarding) {
+///             Welcome()
+///             InterestingModules()
+///             if HKHealthStore.isHealthDataAvailable() {
+///                 HealthKitPermissions()
+///             }
+///             if !localNotificationAuthorization {
+///                 NotificationPermissions()
+///             }
+///             FinalStep()
+///         }
+///         .task {
+///             localNotificationAuthorization = await ...
+///         }
+///     }
+/// }
+/// ```
+///
+/// ### Identifying Navigation Steps
+///
+/// By default, the ``ManagedNavigationStack`` will identify and track the different navigation steps based on their type and location within the stack.
+/// The ``SwiftUICore/View/navigationStepIdentifier(_:)`` modifier allows overriding the default identifier
+/// with a custom one that will be used instead throughout the ``ManagedNavigationStack``.
+/// In most scenarios, the default identifier is sufficient, but there are some edge cases where a custom identifier needs to be specified,
+/// namely when the ``ManagedNavigationStack`` contains multiple steps with the same `View` type, and you wish to programmatically
+/// navigate to one of them, be it as the stack's starting step (via the `startAtStep` parameter in ``ManagedNavigationStack/init(didComplete:path:startAtStep:_:)``,
+/// or via e.g. ``ManagedNavigationStack/Path/navigateToNextStep(matching:includeIntermediateSteps:)``.
+///
+///
+/// ```swift
+/// struct Onboarding: View {
+///     @AppStorage(StorageKeys.onboardingFlowComplete)
+///     var completedOnboardingFlow = false
+///
+///     var body: some View {
+///         ManagedNavigationStack(didComplete: $completedOnboardingFlow) {
+///             MyOwnView()
+///                 .navigationStepIdentifier("step1")
+///             MyOwnView()
+///                 .navigationStepIdentifier("step2")
+///             // Other views as needed
+///         }
+///     }
+/// }
+/// ```
+///
+/// - Note: When the ``SwiftUICore/View/navigationStepIdentifier(_:)`` modifier is applied multiple times to the same view, the outermost identifier takes precedence.
+///
+/// ## Topics
+/// ### Creating a Managed Navigation Stack
+/// - ``init(didComplete:path:startAtStep:_:)``
+/// - ``NavigationFlowBuilder``
+/// ### Navigation
+/// - ``Path``
+/// ### SwiftUI Environment Values
+/// - ``SwiftUICore/EnvironmentValues/isInManagedNavigationStack``
+/// - ``SwiftUICore/EnvironmentValues/isFirstInManagedNavigationStack``
+public struct ManagedNavigationStack: View {
+    static var logger: Logger { Logger(subsystem: "edu.stanford.spezi.SpeziViews", category: "ManagedNavigationStack") }
+    
+    private let navigationFlow: _NavigationFlow
```

Comment:
I really don't like the name `NavigationFlow`, sorry. At least `Collection` must be present in the  name